corehq/apps/export/tests/test_export_data_schema.py

```diff
@@ -292,11 +319,53 @@ def test_basic_application_schema(self):
 
         self.assertEqual(len(schema.group_schemas), 1)
 
+    def test_build_from_saved_schema(self):
+        app = self.current_app
+
+        with patch(
+                'corehq.apps.export.models.new.FormExportDataSchema.get',
+                wraps=FormExportDataSchema.get) as wrapped_get:
+            schema = FormExportDataSchema.generate_schema_from_builds(
+                app.domain,
+                app._id,
+                'my_sweet_xmlns'
+            )
+            self.assertFalse(
+                wrapped_get.called,
+                'Schema should be created from new object'
+            )
+
+        self.assertEqual(len(schema.group_schemas), 1)
+        self.assertEqual(schema.last_app_versions[app._id], 3)
+
+        # After the first schema has been saved let's add a second app to process
+        second_build = Application.wrap(self.get_json('basic_application'))
+        second_build._id = '456'
+        second_build.copy_of = app.get_id
+        second_build.version = 6
+        second_build.save()
+
+        with patch(
+                'corehq.apps.export.models.new.FormExportDataSchema.get',
+                wraps=FormExportDataSchema.get) as wrapped_get:
+
+            new_schema = FormExportDataSchema.generate_schema_from_builds(
+                app.domain,
+                app._id,
+                'my_sweet_xmlns'
+            )
+            wrapped_get.assert_called_with(schema._id)
+
+        second_build.delete()
```

Comment:
If this is for cleanup you can do this to ensure it gets called even if the test fails or there is some other exception:
```python
self.addCleanup(second_build.delete)
```